src/safety_check.c

```diff
@@ -3,6 +3,13 @@
 
 static void (*safety_check_abort)(const char *message);
 
+void safety_check_fail_sized_dealloc(void) {
```

Comment:
The two callers of this are slightly different; in the old one,  the stack trace when it fires is useless (i.e. it may not have been the source of the corruption). In the new one, the place where it fires gives the bug. That might justify a parameter or something to indicate which.

Not a strong opinion, disregard if you disagree.

Reply:
That's a valid concern. Let me tweak it a bit.